service/awsconfig/v10/adapter/adapter.go

```diff
@@ -35,19 +35,20 @@ import (
 type hydrater func(Config) error
 
 type Adapter struct {
-	ASGType          string
-	AvailabilityZone string
-	ClusterID        string
-	MasterImageID    string
-	WorkerImageID    string
-
+	ASGType                    string
+	AvailabilityZone           string
+	ClusterID                  string
+	MasterImageID              string
+	MasterInstanceResourceName string
+	WorkerImageID              string
+
+	Instance       *instanceAdapter
```

Comment:
Am I correct that this is possible to access the same information in 3 ways? And even in 2 ways from the same object: `Adapter.MasterInstanceResourceName` and `Adapter.Instance.Master.Instance.ResourceName`?

Reply:
When you point is that this has to be refactored over time then you are right.